std/algorithm/comparison.d

```diff
@@ -604,85 +604,85 @@ Returns:
 
 */
 int cmp(alias pred = "a < b", R1, R2)(R1 r1, R2 r2)
-if (isInputRange!R1 && isInputRange!R2 && !(isSomeString!R1 && isSomeString!R2))
+if (isInputRange!R1 && isInputRange!R2 || isSomeString!R1 && isSomeString!R2)
```

Comment:
Yeah, you have to be very careful with `isSomeString` in template constraints for range-based functions, because enums pass `isSomeString` but are not ranges.